apps/codeforafrica/src/components/NavBar/MobileNavigation/index.js

```diff
@@ -0,0 +1,100 @@
+import { Grid, Dialog, DialogContent, Slide } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import PropTypes from "prop-types";
+import React from "react";
+
+/* eslint-disable import/no-unresolved */
+import cfaLogo from "@/codeforafrica/assets/CfA logo.svg?url";
+import menuIcon from "@/codeforafrica/assets/menu-icon.svg?url";
+import ImageIcon from "@/codeforafrica/components/ImageIcon";
+import NavBarNavList from "@/codeforafrica/components/NavBarNavList";
+
+const DialogContainer = styled(Dialog)(({ theme: { palette, spacing } }) => ({
+  "& .MuiDialog-container": {
+    height: "100%",
+  },
+  "& .MuiBackdrop-root": {
+    background: "transparent",
+  },
+  "& .MuiDialogContent-root": {
+    padding: spacing(2),
+    color: palette.text.secondary,
+    background: palette.primary.main,
+  },
+}));
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="down" ref={ref} {...props} />;
+});
+
+const MobileNavigation = React.forwardRef(function MobileNavigation(
+  props,
+  ref
+) {
+  const { menu } = props;
+
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <Grid
+      container
+      justifyContent="space-between"
+      alignItems="center"
+      direction="row"
+      ref={ref}
+      sx={{ display: { xs: "flex", md: "none" } }}
+    >
+      <Grid item xs={4} sx={{ ml: 2, py: 1 }}>
+        <ImageIcon src={cfaLogo} alt="Logo" width="116px" height="50px" />
+      </Grid>
+      <Grid item xs={6} container justifyContent="flex-end" sx={{ mr: 2 }}>
```

Comment:
Similarly here `<Grid item` should be enough and let everything take it's natural size.